### PR TITLE
SDIT-1174 CaseNoteResource

### PR DESCRIPTION
**Pull request by steverendell**
Add PreAuthorize roles to /case-notes/*

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderResource.java [hunk 1]
```diff
@@ -471,7 +471,7 @@ public PrisonerIdentifier getNextPrisonerIdentifier() {
         @ApiResponse(responseCode = "400", description = "Invalid request.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))}),
         @ApiResponse(responseCode = "404", description = "Requested resource not found.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))}),
         @ApiResponse(responseCode = "500", description = "Unrecoverable error occurred whilst processing request.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))})})
-    @Operation(summary = "Return a set Incidents for a given offender No.", description = "Can be filtered by participation type and incident type")
+    @Operation(summary = "Return a set Incidents for a given offender No.", description = "Can be filtered by participation type and incident type. Requires the VIEW_INCIDENTS role.")
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderResource.java:474]: tweak from earlier change

File: src/main/java/uk/gov/justice/hmpps/prison/repository/CaseNoteRepository.java [hunk 2]
```diff
@@ -83,16 +74,6 @@ public List<CaseNoteUsage> getCaseNoteUsage(@NotNull final LocalDate fromDate, @
...
-    public List<CaseNoteEvent> getCaseNoteEvents(final LocalDateTime fromDate, final Set<String> events, final long limit) {
-        final var casenoteevents = jdbcTemplate.query(queryBuilderFactory.getQueryBuilder(CaseNoteRepositorySql.RECENT_CASE_NOTE_EVENTS.getSql(), Map.of()).addPagination().build(),
-            createParamSource(new PageRequest(0L, limit),
-                "fromDate", new SqlParameterValue(Types.TIMESTAMP, fromDate),
-                "types", events),
-            CASE_NOTE_EVENT_ROW_MAPPER);
-        return casenoteevents.stream().map(CaseNoteEventDto::toCaseNoteEvent).collect(Collectors.toList());
-    }
-
-
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/CaseNoteRepository.java:95]: deleted 3 case-notes endpoints as they were unused

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/CaseNoteResource.java [hunk 3]
```diff
@@ -72,73 +59,61 @@ public List<CaseNoteStaffUsage> getCaseNoteStaffUsageSummaryByPost(@RequestBody
...
         return caseNoteService.getCaseNoteUsage(type, subType, offenderNo, staffId, agencyId, fromDate, toDate, ObjectUtils.defaultIfNull(numMonths, 1));
     }
 
     @ApiResponses({
             @ApiResponse(responseCode = "200", description = "The case note usage list is returned.")})
-    @Operation(summary = "Retrieves list of case notes grouped by type and offender", description = "Retrieves list of case notes grouped by type and offender")
+    @Operation(summary = "Retrieves list of case notes grouped by type and offender", description = "Retrieves list of case notes grouped by type and offender. Requires role VIEW_CASE_NOTES")
     @PostMapping("/usage")
     @SlowReportQuery
+    @PreAuthorize("hasRole('VIEW_CASE_NOTES')")
```
Comment (karenmillermoj) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/CaseNoteResource.java:84]: just checking that the clients calling this now have the VIEW_CASE_NOTES role (if they didn't already) possibly hmpps-prisoner-profile-system etc?

**Review by karenmillermoj — APPROVED** [review]
lgtm

**Pull request merged**